engine/table/src/main/java/io/deephaven/engine/table/impl/by/PartitionByChunkedOperator.java

```diff
@@ -446,10 +519,13 @@ private boolean appendShifts(@NotNull final LongChunk<? extends RowKeys> preShif
     @Override
     public void ensureCapacity(final long tableSize) {
         tables.ensureCapacity(tableSize);
-        addedRowSets.ensureCapacity(tableSize);
+        addedBuilders.ensureCapacity(tableSize);
```

Comment:
The way you are currently doing things, `addedBuilders` is only ever needed for refreshing parents, and should be moved into the appropriate `if` block. That said, I am advocating some consolidation.